src/openforms/registrations/contrib/zgw_apis/utils.py

```diff
@@ -0,0 +1,36 @@
+from datetime import date, datetime
+
+import pytz
+
+
+def process_according_to_eigenschap_format(specificatie: dict, value: str):
+    """
+    This is an extra validation we are doing against the eigenschap.specificatie.formaat
+    since the value we send should be YYYYMMDD for date or YYYYMMDDHHmmSS for date/time.
+    """
+    format = specificatie.get("formaat")
+    if format not in ["datum", "datum_tijd"]:
+        return value
+
+    if format == "datum":
+        try:
+            valid_date = date.fromisoformat(value)
+        except ValueError as exc:
+            raise exc
```

Comment:
since you just re-raise, this could instead be:

```suggestion
        valid_date = date.fromisoformat(value)
```